File: M3Tools/Dialogs/Emails/EmailBodySelection.vb
Imports System.Windows.Forms

Namespace Dialogs
	Public Class EmailBodySelection
		Private ReadOnly Property Subject As String
			Get
				Select Case TabControl1.SelectedIndex
					Case tp_Templates.TabIndex
						Return ts_Templates.TemplateSubject
					Case tp_Custom.TabIndex
						Return CustomEmail1.Subject
					Case Else
						Return Nothing
				End Select
			End Get
		End Property

		Private ReadOnly Property Body As String
			Get
				Select Case TabControl1.SelectedIndex
					Case tp_Templates.TabIndex
						Return ts_Templates.TemplateValue
					Case tp_Custom.TabIndex
						Return CustomEmail1.Body
					Case Else
						Return Nothing
				End Select
			End Get
		End Property

		Private ReadOnly Property BodyType As Types.GTools.EmailType
			Get
				Select Case TabControl1.SelectedIndex
					Case tp_Templates.TabIndex
						Return Types.GTools.EmailType.HTML
					Case tp_Custom.TabIndex
						Return Types.GTools.EmailType.PLAIN
					Case Else
						Return Nothing
				End Select
			End Get
		End Property

		ReadOnly Property Content As Types.GTools.EmailContent
			Get
				Return New Types.GTools.EmailContent(Subject, Body, BodyType)
			End Get
		End Property

		WriteOnly Property Templates As Types.TemplateList
			Set(templates As Types.TemplateList)
				ts_Templates.AddRange(templates)
			End Set
		End Property


		Private Sub FinishDialog(sender As Object, e As EventArgs) Handles OK_Button.Click
			Me.DialogResult = DialogResult.OK
			Me.Close()
		End Sub

		Private Sub CancelDialog(sender As Object, e As EventArgs) Handles Cancel_Button.Click
			Me.DialogResult = DialogResult.Cancel
			Me.Close()
		End Sub

		Private Sub Reload() Handles MyBase.Load
			ts_Templates.Reload()
		End Sub
	End Class
End Namespace

File: M3Tools/Dialogs/Emails/EmailBodySelection.cs
using System;
using System.Windows.Forms;

using SPPBC.M3Tools.Types;

namespace SPPBC.M3Tools.Dialogs
{
	/// <summary>
	/// A control for selecting from templates for email body
	/// </summary>
	public partial class EmailBodySelection
	{
		private string Subject
		{
			get
			{
				return tc_EmailType.SelectedIndex switch
				{
					var @template when @template == tc_EmailType.TabPages.IndexOf(tp_Templates) => ts_Template.TemplateSubject,
					var @custom when @custom == tc_EmailType.TabPages.IndexOf(tp_Custom) => ce_Custom.Subject,
					_ => null,
				};
			}
		}

		private string Body
		{
			get
			{
				return tc_EmailType.SelectedIndex switch
				{
					var @template when @template == tc_EmailType.TabPages.IndexOf(tp_Templates) => ts_Template.TemplateValue,
					var @custom when @custom == tc_EmailType.TabPages.IndexOf(tp_Custom) => ce_Custom.Body,
					_ => null,
				};
			}
		}

		/// <summary>
		/// The content of the email being sent
		/// </summary>
		public Types.GTools.EmailContent Content => new(Subject, Body);

		/// <summary>
		/// <inheritdoc/>
		/// </summary>
		public EmailBodySelection(TemplateList templates)
		{
			InitializeComponent();
			ts_Template.AddRange(templates);

		}

		private void ConfirmSelection(object sender, EventArgs e)
		{
			DialogResult = DialogResult.OK;
			Close();
		}

		private void CancelSelection(object sender, EventArgs e)
		{
			DialogResult = DialogResult.Cancel;
			Close();
		}

		private void Reload(object sender, EventArgs e) => ts_Template.Reload();
	}
}
